protobuf/syft_proto/execution/v1/type_wrapper.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package syft_proto.execution.v1;
+option java_package = "org.openmined.syftproto.execution.v1";
+
+message ClassType {
+    /*
+        A type contains the serialization id and string bytes.
+    */
+    int32 id = 1;
+    bytes type = 2;
+}
+
+message NestedTypeWrapper {
+    /*
+    Message to bufferize a nested type. In a nested type we can have:
+        1. A type.
+        2. A list of nested types.
+        3. A tuple of nested types.
+        4. A dict of nested types.
+    */
+
+    // Message defined to simulate the behavior of a generic map.
+    message KeyValue {
+        // bytes to serialize the key, currently only str is supported.
+        bytes key = 1;
+        NestedTypeWrapper value = 2;
+    }
+
+    // To create the behavior described above, we need a new message, as
+    // oneof will not permit repeated fields.
+    message TypeContainer {
+        oneof nested_container {
+            // First scenario, acts like a list with one element, the type.
+            ClassType nested_type = 1;
+
+            // Second scenario, a list of possibly nested elements.
+            NestedTypeWrapper nested_type_list = 2;
+
+            // Third scenario, a tuple of possibly nested elements.
+            NestedTypeWrapper nested_type_tuple = 3;
+
+            // Last scenario, a dict of possible nested elements.
+            KeyValue nested_type_dict = 4;
```

Comment:
Could this be a `map<string, NestedTypeWrapper>`?

Reply:
I made this approach to simulate a generic map, but not sure how to do it properly yet (currently support only strings, but should be any type from the supported by our protobuf schemas).

Reply:
Why do we need non-string keys? What would we use them for?

Reply:
Good question, I don't think I have an answer. I'll leave it as a string.